Sources/SmileID/Classes/DocumentVerification/Model/OrchestratedDocumentVerificationViewModel.swift

```diff
@@ -118,117 +117,89 @@ internal class IOrchestratedDocumentVerificationViewModel<T, U: JobResult>: Obse
-                    jobId: authResponse.partnerParams.jobId,
-                    includeImageLinks: false,
-                    includeHistory: false,
-                    timestamp: authResponse.timestamp,
-                    signature: authResponse.signature
-                )
-                return SmileID.api.getJobStatus(request: jobStatusRequest)
-            }
-            .sink(
-                receiveCompletion: { completion in
-                    switch completion {
-                    case .failure(let error):
-                            // todo will updated this in a subsequent PR to use SmileID.allowOfflineMode property
-                            // we need to check that property and figure our if we need to save in unsubmitted folder
-                            // or move to submitted folder - I will update the documentation as well
-                            do {
-                                _ = try LocalStorage.saveOfflineJob(
-                                    allowOfflineMode: SmileID.allowOfflineMode,
-                                    jobId: self.jobId,
-                                    userId: self.userId,
-                                    jobType: self.jobType,
-                                    enrollment: false,
-                                    allowNewEnroll: self.allowNewEnroll,
-                                    partnerParams: self.extraPartnerParams
-                                )
-                            } catch {
-                                print("Error saving job for offline mode: \(error)")
-                                self.onError(error: error)
-                            }
-                        print("Error submitting job: \(error)")
-                        self.onError(error: error)
-                    default:
-                        break
-                    }
-                },
-                receiveValue: { response in
-                    self.jobStatusResponse = response
-                    do {
-                        try LocalStorage.moveToSubmittedJobs(jobId: self.jobId)
-                    } catch {
-                        print("Error moving job to submitted directory: \(error)")
-                        self.onError(error: error)
-                    }
-                    DispatchQueue.main.async {
-                        self.step = .processing(.success)
-                    }
+                let prepUploadResponse = try await SmileID.api.prepUpload(request: prepUploadRequest).async()
+                _ = try await SmileID.api.upload(
+                    zip: zip,
+                    to: prepUploadResponse.uploadUrl
+                ).async()
+                didSubmitJob = true
+                do {
+                    try LocalStorage.moveToSubmittedJobs(jobId: self.jobId)
+                } catch {
+                    print("Error moving job to submitted directory: \(error)")
+                    self.onError(error: error)
```

Comment:
Do we need to add a return here?
```suggestion
                    self.onError(error: error)
                    return
```